### PR TITLE
Task#11943 Share Screen

### PR DESCRIPTION
**Pull request by huybnd-1817**
### Issue Link :link:
https://edu-redmine.sun-asterisk.vn/issues/11943

### Goals :soccer:
Create UI for share screen
- Image's achievement
- User's thought
- Share button
- Back button

Create a UI action
- Click share button to share this post via social media
- Back to previous screen

### Implementation Details :construction:
https://youtu.be/OQDe8ZnW07Q

### CODE REVIEW

File: CuisineRecipe/CuisineRecipe/Scenes/DetailRecipe/DetailRecipeViewController.swift [hunk 1]
```diff
@@ -178,6 +190,66 @@ extension DetailRecipeViewController: UICollectionViewDataSource {
...
+                self.selectImageFrom(.camera)
+            } else {
+                self.showAlert(message: ErrorMessages.cameraNotAvailable)
+            }
+        }))
+        
+        //ActionSheet Photo Library
+        actionSheet.addAction(UIAlertAction(title: ImageSource.photoLibrary.rawValue,
+                                            style: .default,
+                                            handler: { _ in
```
Comment (dongsonbk58) [inline CuisineRecipe/CuisineRecipe/Scenes/DetailRecipe/DetailRecipeViewController.swift]: [weak self] này chú

File: CuisineRecipe/CuisineRecipe/Scenes/DetailRecipe/DetailRecipeViewController.swift [hunk 2]
```diff
@@ -131,6 +135,14 @@ final class DetailRecipeViewController: UIViewController {
         recipe?.addToIngredients(ingredients as NSSet)
         showToast(message: ToastMessage.addRecipeSuccessful)
     }
+    
+    @IBAction private func handleSnapTapped(_ sender: Any) {
+        if checkPermission(completion: { message in
```
Comment (dongsonbk58) [inline CuisineRecipe/CuisineRecipe/Scenes/DetailRecipe/DetailRecipeViewController.swift]: [weak self]

File: CuisineRecipe/CuisineRecipe/Scenes/DetailRecipe/DetailRecipeViewController.swift [hunk 3]
```diff
@@ -178,6 +190,66 @@ extension DetailRecipeViewController: UICollectionViewDataSource {
...
 }
 
+extension DetailRecipeViewController: UIImagePickerControllerDelegate, UINavigationControllerDelegate {
+    fileprivate func setupActionSheet() {
+        let actionSheet = UIAlertController(title: nil, message: nil, preferredStyle: .actionSheet)
+        
+        //ActionSheet Camera
+        actionSheet.addAction(UIAlertAction(title: ImageSource.camera.rawValue,
+                                            style: .default,
+                                            handler: { _ in
```
Comment (dongsonbk58) [inline CuisineRecipe/CuisineRecipe/Scenes/DetailRecipe/DetailRecipeViewController.swift]: đây nữa.. xem lại toàn bộ project nhé

File: CuisineRecipe/CuisineRecipe/Scenes/Share/ShareViewController.swift [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        super.viewDidLoad()
+        configView()
+    }
+    
+    private func configView() {
+        hideKeyboardWhenTappedAround()
+        guard let myImage = achievementImage else { return }
+        achievementImageView.image = myImage
+        
+        descriptionTextView.text = Constants.descriptionPlaceholder
```
Comment (dongsonbk58) [inline CuisineRecipe/CuisineRecipe/Scenes/Share/ShareViewController.swift]: suggest dùng then cho gọn nhé

**Review by tienpx3010 — APPROVED** [review]

**Pull request merged**